_drafts/A BLE Primer.md

```diff
@@ -0,0 +1,1165 @@
+- Define a GATT service which we'll call "simple service" which consists of one characteristic:
+	- A "**Button 1**" Characteristic (exposes the current state of the development kit's Button 1: pressed (0x01) or released (0x00)). The Characteristic will allow both **reads** and **notifications** to allow a client to read the value and get notified only when the value changes (instead of polling for the value waiting for a change to occur). 
+- Handling of the Button 1 events (presses and releases).
+
+Let's go through the steps and code changes needed to implement our application. All the code used in this blog post is available on [GitHub](https://github.com/memfault/interrupt/tree/master/example/ble-primer/).
+
+#### Enabling the nRF Log Module
+
+In order to enable logging and see debug data, you need to do the following (some of these may already be enabled in your template application) in `pca10056/s140/config/sdk_config.h`:
+
+- Enable the `NRF_LOG_ENABLED` macro:
+
+	```
+	// <e> NRF_LOG_ENABLED - nrf_log - Logger
+	//==========================================================
+	#ifndef NRF_LOG_ENABLED
+	#define NRF_LOG_ENABLED 1
+	#endif
+	```
+	
+- Set the log level to 3 (or 4 more debugging information):
+
+	```
+	// <o> NRF_LOG_DEFAULT_LEVEL  - Default Severity level
+	 
+	// <0=> Off 
+	// <1=> Error 
+	// <2=> Warning 
+	// <3=> Info 
+	// <4=> Debug 
+	
+	#ifndef NRF_LOG_DEFAULT_LEVEL
+	#define NRF_LOG_DEFAULT_LEVEL 3
+	#endif
+	```
+	
+- Since we'll be using a terminal program (such as CoolTerm), we need to enable logging via the UART backend. This is done by enabling the `NRF_LOG_BACKEND_UART_ENABLED` as follows:
+
+	```
+	// <e> NRF_LOG_BACKEND_UART_ENABLED - nrf_log_backend_uart - Log UART backend
+	//==========================================================
+	#ifndef NRF_LOG_BACKEND_UART_ENABLED
+	#define NRF_LOG_BACKEND_UART_ENABLED 1
+	#endif
+	```
+	
+That's it! Now you have logging enabled and you will be able to see log messages show up in your terminal program of choice. We'll get to configuring the terminal program in the "Testing" section.
+
+#### Handling the Button Presses
+
+The application already comes ready with code that sets up the LEDs and Buttons to be used. However, we'll be modifying it to utilize an nRF module called the "App Button" module. This module simplifies the code for handling button events compared to the originally included BSP library.
+
+You can find more about the App Button library here:
+
+[**nRF5 SDK Button Handler Library**](https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.3.0/group__app__button.html?resultof=%22%61%70%70%22%20%22%62%75%74%74%6f%6e%22%20)
+
+We will need to add three different segments of code to `main.c` to enable handling of Button 1 events:
+
+- Configure the button we'll be using (Button 1 on the dev kit). We also add a forward declaration of the function responsible for handling the button events: 
+
+	```
```

Comment:
Can add `c` to these so they are formatted as code in the blog post

Reply:
Done